core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -178,6 +182,9 @@ class Qualification(outputPath: String, numRows: Int, hadoopConf: Configuration,
             val newClusterSummary = tempSummary.clusterSummary.copy(
               recommendedClusterInfo = pluginTypeChecker.platform.recommendedClusterInfo)
             val newQualSummary = tempSummary.copy(clusterSummary = newClusterSummary)
+            if (app.isOlderAttempt) {
+              throw OlderAttemptException(app)
+            }
```

Comment:
This could be simpler.
`allApps` is a concurrentSet. We could simply use that to detect failed attempts before an app is added to the concurrent set and throw an exception.